opentracing/dapperish/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"html"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+
+	"github.com/opentracing/api-golang/opentracing"
+)
+
+func client() {
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		span, ctx := opentracing.StartSpan("getInput")
+		span.Info("ctx: ", ctx)
+		fmt.Print("Enter text: ")
+		text, _ := reader.ReadString('\n')
+		span.Info(text)
+
+		httpClient := &http.Client{}
+		httpReq, _ := http.NewRequest("POST", "http://localhost:8080/", bytes.NewReader([]byte(text)))
+		opentracing.AddContextIDToHttpHeader(span.ContextID(), httpReq.Header)
+		resp, err := httpClient.Do(httpReq)
+		if err != nil {
+			span.Error("error: ", err)
+		} else {
+			span.Info("got response: ", resp)
+		}
+
+		span.Finish()
+	}
+}
+
+func server() {
+	http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		reqCtxID, err := opentracing.GetContextIDFromHttpHeader(
+			req.Header, opentracing.GlobalTracer())
+		if err != nil {
+			panic(err)
+		}
+
+		serverSpan, _ := opentracing.StartSpan("serverSpan", reqCtxID)
```

Comment:
I'd like that, but how does it work with clock skew adjustment? Do we need need to capture additional information in that case?

Someone asked me a related question the other day. We're using Hyperbahn routing mesh as discovery mechanism, so an RPC call has two hops through the routers ring before reaching the server. The question was how these two hops can be represented as their own spans. In the current Zipkin model they cannot be, since client/server are a single span. If we split those, then I think the client span could have three siblings, {hop1, hop2, server}. But I am not clear how to express that in this new API, or how to do clock skew adjustment without knowing exact causality of these spans, which we're not capturing.